ui/ducks/send/send.test.js

```diff
@@ -1,534 +1,1544 @@
-      store.dispatch(updateSendEnsResolution('0x00'));
-      const state = store.getState();
-      expect(state.send.ensResolutionError).toBe('');
-      expect(state.send.ensResolution).toBe('0x00');
+        expect(result.draftTransaction.txParams.to).toStrictEqual(
+          action.payload.address,
+        );
+        expect(result.draftTransaction.txParams.value).toStrictEqual(
+          action.payload.amount,
+        );
+        expect(result.draftTransaction.txParams.gasPrice).toStrictEqual(
+          action.payload.gasPrice,
+        );
+        expect(result.draftTransaction.txParams.gas).toStrictEqual(
+          action.payload.gasLimit,
+        );
+      });
     });
 
-    it('updateSendEnsResolutionError clears ensResolution and sets ensResolutionError', () => {
-      store.dispatch(updateSendEnsResolution('0x00'));
-      store.dispatch(updateSendEnsResolutionError('notFound'));
-      const state = store.getState();
-      expect(state.send.ensResolutionError).toBe('notFound');
-      expect(state.send.ensResolution).toBeNull();
+    describe('updateGasLimit', () => {
+      const action = {
+        type: 'send/updateGasLimit',
+        payload: '0x5208', // 21000
+      };
+
+      it('should', () => {
+        const result = sendReducer(initialState, action);
+
+        expect(result.gas.gasLimit).toStrictEqual(action.payload);
+        expect(result.draftTransaction.txParams.gas).toStrictEqual(
+          action.payload,
+        );
+      });
+
+      it('should recalculate gasTotal', () => {
+        const gasState = {
+          ...initialState,
+          gas: {
+            gasLimit: '0x0',
+            gasPrice: '0x3b9aca00', // 1000000000
+          },
+        };
+
+        const result = sendReducer(gasState, action);
+
+        expect(result.gas.gasLimit).toStrictEqual(action.payload);
+        expect(result.gas.gasPrice).toStrictEqual(gasState.gas.gasPrice);
+        expect(result.gas.gasTotal).toStrictEqual('0x1319718a5000'); // 21000000000000
+      });
     });
 
-    it('resetSendState sets state back to the initial value', () => {
-      store.dispatch(setGasLimit('0xEE'));
-      store.dispatch(setGasPrice('0xFF'));
-      let state = store.getState();
-      expect(state.send.gasLimit).toBe('0xEE');
-      expect(state.send.gasPrice).toBe('0xFF');
-      store.dispatch(resetSendState());
-      state = store.getState();
-      expect(state.send).toMatchObject(initialState);
+    describe('updateGasPrice', () => {
+      const action = {
+        type: 'send/updateGasPrice',
+        payload: '0x3b9aca00', // 1000000000
+      };
+
+      it('should', () => {
+        const result = sendReducer(initialState, action);
+
+        expect(result.gas.gasPrice).toStrictEqual(action.payload);
+        // expect(result.draftTransaction.txParams.gasPrice).toStrictEqual(action.payload) // TODO: Failing.Shouldnt this also update the draftTransaction?
```

Comment:
Check

Reply:
It should, as long as the entire send state is valid. if it isn't valid it doesn't update the draft.